### PR TITLE
[CSS4Pseudo] Added 9 tests, 7 references and 2 support files

### PR DESCRIPTION
**Pull request by TalbotG**
This PR is a followup to 
https://github.com/web-platform-tests/wpt/pull/18213
and it has all the corrections requested and suggestions proposed in it. 

active-selection-003-manual.html
active-selection-011.html
active-selection-011-ref.html
active-selection-012.html
active-selection-012-ref.html
active-selection-014.html
active-selection-014-ref.html
active-selection-015.html
active-selection-015-ref.html
active-selection-016.html
active-selection-016-ref.html
active-selection-017.html
active-selection-017-ref.html
active-selection-018.html
active-selection-019.html
active-selection-019-ref.html
support/select-custom.png
support/select-custom.cur

### CODE REVIEW

**Comment by TalbotG** [comment]
@fantasai , can you review this please?

**Comment by fantasai** [comment]
OK, replying to your comments in #18213 here, then.

> I can remove or drop active-selection-001-manual.html (testing color) active-selection-002-manual.html (testing background-color) active-selection-004-manual.html (testing text-decoration). No problem.
>But why should I remove
active-selection-003-manual.html (testing cursor) ?

I don't think I suggested to remove any of these tests? I just suggested that they are too focused on checking for the one thing that changed, but they should check other side effects as well.

For example:

- 001 doesn't pass only when color of the text is green when selected, but also there must be no background color for the selection.
- 002 needs to require not only that the background is green, but that the text color stays black
- 003 needs to check not only that the cursor changes, but that the background color and color of the selected sample text matches the same background color and color of the instructions when they are highlighted. E.g. Firefox must fail this test because it does not show the highlight at all!
- 004 same as 003, pass condition needs to check that normal selection color + bg color are still in effect

> As far as I remember, active-selection-18 is just a bit more complex than 011 and 012. But maybe I kept active-selection-018 for additional reasons... I can not remember if this is the case...
For example, why the test declared div::selection { background-color: white; } in that 018 test is not clear in my mind right now and could be related to some difficulty I had... There has to be a reason why I declared div::selection { background-color: white; } and I can not remember it right now...

This is why meta assert and/or unique titles is useful. :) If we're not sure why we're adding, maybe don't add?

> >   (Another option would be to test partial selection in the basic tests. You could do this by setting the color of the text and its background to match, making the selected text “PASS”, and coloring it green to make it visible.)
>
> Okay. I see what you are saying. I could be testing partial selection with a "PASS" being at [beginning | middle | end] of 2 "FAIL" that would be white color on white background and only the "PASS" would become green once selected. There can be more variations like that...

What I was saying is maybe all of the tests should be partial-selection tests, then there's only one set of tests instead of separate one for full selection and separate one for partial-selection. Let me know if you think that's impractical.

**Comment by TalbotG** [comment]
> > I can remove or drop active-selection-001-manual.html (testing color) active-selection-002-manual.html (testing background-color) active-selection-004-manual.html (testing text-decoration). No problem.

> I don't think I suggested to remove any of these tests?

I misunderstood you.

>   * 001 doesn't pass only when color of the text is green
> when selected, but also there must be no background 
> color for the selection.

I added your description of the additional side effect to be verified as a pass condition. I also tuned the instructions text to be more precise. Improved test:
http://www.gtalbot.org/BrowserBugsSection/CSS4Pseudo/active-selection-001-manual-new.html

>  * 002 needs to require not only that the background is
> green, but that the text color stays black

I added your description of the additional side effect to be verified as a pass condition. I also tuned the instructions text to be more precise. Improved test:
http://www.gtalbot.org/BrowserBugsSection/CSS4Pseudo/active-selection-002-manual-new.html

 
>  * 003 needs to check not only that the cursor changes, 
> but that the background color and color of the selected 
> sample text matches the same background color and 
> color of the instructions when they are highlighted. E.g. 
> Firefox must fail this test because it does not show the 
> highlight at all!

You are correct. The background color and color of the selected sample text needed to be verified and need to match UA-default colors for selection. Improved test (maybe a bit too wordy):
http://www.gtalbot.org/BrowserBugsSection/CSS4Pseudo/active-selection-003-manual-new.html


>  * 004 same as 003, pass condition needs to check that
> normal selection color + bg color are still in effect

Improved test:
http://www.gtalbot.org/BrowserBugsSection/CSS4Pseudo/active-selection-004-manual-new.html

> > As far as I remember, active-selection-18 is just a bit more complex than 011 and 012. But maybe I kept active-selection-018 for additional reasons... I can not remember if this is the case...
> > For example, why the test declared div::selection { background-color: white; } in that 018 test is not clear in my mind right now and could be related to some difficulty I had... There has to be a reason why I declared div::selection { background-color: white; } and I can not remember it right now...

There is (or was?) now a comment explaining the `div::selection { background-color: white; }`
in the active-selection-18 test.

> > > (Another option would be to test partial selection in the basic tests. You could do this by setting the color of the text and its background to match, making the selected text “PASS”, and coloring it green to make it visible.)

> > (...) testing partial selection with a "PASS" being at [beginning | middle | end] of 2 "FAIL" that would be white color on white background and only the "PASS" would become green once selected.

active-selection-015.html
active-selection-016.html
active-selection-017.html
tests which are in the current PR do that.

> What I was saying is maybe all of the tests should be partial-selection tests, then there's only one set of tests instead of separate one for full selection and separate one for partial-selection. Let me know if you think that's impractical.

I think there should be a minority (say, 20%) of tests testing partial selection and a majority of tests testing full selection... like it is now.

In a few min. I will reinstate 
active-selection-001-manual
active-selection-002-manual
active-selection-004-manual
along with your proposed improvements and also improve 
active-selection-003-manual in this current PR.

**Comment by fantasai** [comment]
Suggest some improvements to the instructions for clarity / conciseness:

```
Instructions: Select a few or all characters from the words "Text sample" using a variety of methods:
<ul>
  <li>Select by dragging the mouse across the words.
  <li>Select by using shift + keyboard arrows when keyboard navigation / caret browsing is enabled.
  <li>Double click a word to select it.
  <li>Triple click the text to select it.
</ul>

Test passes if, in every case, the ....
```

Also for 003/4:

```
"Text sample" <em>and</em> the background color and color of the selected text are the same as when selecting text from these instructions.
```

**Comment by fantasai** [comment]
active-selection-011 instructions still need to check the background color (which should be white, not the OS default, for example)
active-selection-012 instructions need to check the text color for the same reason (and maybe that color should be something interesting like fuchsia to make sure it is not changed when highlighted)

active-selection-014 needs to check that the OS default colors are used, right? Adding underline shouldn't remove colors.

**Comment by fantasai** [comment]
There are three tests with the title “CSS Pseudo-Elements Test: active selection of partial text and color” and the assert “"This test checks active selection on a part of text.”. The metadata here should be unique to the test such that it explains what this test is trying to do and how it is testing something different from the others.

**Comment by TalbotG** [comment]
@fantasai , your review and comments are excellent. And your suggestions are welcomed and appreciated. 
I missed a lot in those tests. I will fix several tests {active-selection-011, active-selection-012, active-selection-014} today in this PR 20800.

Also (note to self): 
- the `<link rel="match">` to reference of active-selection-015.html must be uncommented
- a blank line should be added at EOF of active-selection-019-ref.html (Why the lint did not report that? Why none of the 10 Continuous Integration checks report that?)
- in active-selection-014 , the color of the underline must be the color of the OS-default highlight color (which is currentcolor)

**Comment by TalbotG** [comment]
@fantasai , I am closing this PR. Please visit https://github.com/web-platform-tests/wpt/pull/21928 instead, which contains all of the corrections and improvements your review here proposed. Thank you.
